### PR TITLE
[#15] PostgreSQL database and CI environment

### PR DESCRIPTION
**Pull request by lancelafontaine**
[#15]

- [x] Updates docs in all READMEs for how to run tests and lints for a particular service
- [x] Adds a CI `docker-compose` to simplify running tests and lints
- [x] Sets up a Docker network for ability to resolve hostnames across Docker containers
- [x] Sets up a Docker volume for database persistence when Docker containers are teared down
- [x] Creates a PostgreSQL database as a service provided by a container
- [x] Develops a proof-of-concept for running queries against the database from the backend service while both are running in separate containers using `docker-compose`.

### CODE REVIEW

**Review by dodoels — APPROVED** [review]

File: docker-compose.ci.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    command: bash -c "pytest && flake8"
+  nlp_service:
+    command: bash -c "pytest && flake8"
+  backend_service:
+    command: bash -c "pytest && flake8"
+  web_client:
+    command: bash -c "npm run test && npm run lint"
+  postgresql_db:
+    environment:
+      POSTGRES_PASSWORD: DEV_PASS_NOT_SECRET
```
Comment (arekmano) [inline docker-compose.ci.yml:14]: What about our prod pass?

File: src/postgresql_db/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# PostgreSQL Database
```
Comment (arekmano) [inline src/postgresql_db/README.md:1]: I just want to make sure that PROD data is not ephemeral; we most certainly need user input to further train our classifiers.

**Review by arekmano — CHANGES_REQUESTED** [review]
I would like my prod-related question answered before merge.

File: docker-compose.ci.yml (hunk 1 above)
Comment (lancelafontaine) [reply docker-compose.ci.yml:14]: Was thinking of GPG encrypted password decryptes with a key that is only the on prod machine.

File: src/postgresql_db/README.md (hunk 2 above)
Comment (lancelafontaine) [reply src/postgresql_db/README.md:1]: Dev data is not ephemeral either. It's using a volume to store on your local. For prod data, we will have to include some kind of db backup strategy.

**Review by arekmano — APPROVED** [review]
Approved but I would like some form of plan for the prod database

**Comment by lancelafontaine** [comment]
Will do. I'll start Prod configuration soon. I'll propose something, but I'll also speak with everyone on the team to determine the needs for the db for all services. It will give me a better idea of what is required.

**Pull request merged**